ui/dashboards/src/context/DashboardProvider/layout-slice.tsx

```diff
@@ -190,3 +222,20 @@ function getYForNewRow(group: PanelGroupDefinition) {
   }
   return newRowY;
 }
+
+// Return an object that maps each panel to the groups it belongs
+function mapPanelGroupToPanel(groups: PanelGroupDefinition[]) {
```

Comment:
What do you think of `mapPanelToPanelGroups`? Since the mapping is from a panel key to a list of panel group IDs.